ports/esp32/adc.c

```diff
@@ -33,30 +33,27 @@
 #define DEFAULT_VREF 1100
 
 void madcblock_bits_helper(machine_adc_block_obj_t *self, mp_int_t bits) {
-    switch (bits) {
-        #if CONFIG_IDF_TARGET_ESP32
-        case 9:
-            self->width = ADC_WIDTH_BIT_9;
-            break;
-        case 10:
-            self->width = ADC_WIDTH_BIT_10;
-            break;
-        case 11:
-            self->width = ADC_WIDTH_BIT_11;
-            break;
-        #endif
-        #if CONFIG_IDF_TARGET_ESP32 || CONFIG_IDF_TARGET_ESP32C3 || CONFIG_IDF_TARGET_ESP32C6 || CONFIG_IDF_TARGET_ESP32S3
-        case 12:
-            self->width = ADC_WIDTH_BIT_12;
-            break;
-        #endif
-        #if CONFIG_IDF_TARGET_ESP32S2
-        case 13:
-            self->width = ADC_WIDTH_BIT_13;
-            break;
-        #endif
-        default:
-            mp_raise_ValueError(MP_ERROR_TEXT("invalid bits"));
+    if (bits >= SOC_ADC_RTC_MIN_BITWIDTH && bits <= SOC_ADC_RTC_MAX_BITWIDTH) {
+        switch (bits) {
+            case 9:
+                self->width = ADC_BITWIDTH_9;
+                break;
+            case 10:
+                self->width = ADC_BITWIDTH_10;
+                break;
+            case 11:
+                self->width = ADC_BITWIDTH_11;
+                break;
+            case 12:
+                self->width = ADC_BITWIDTH_12;
+                break;
+            case 13:
+                self->width = ADC_BITWIDTH_13;
+                break;
+            default:
```

Comment:
Should this default do anything, raise an exception, or just not exist?  Is it possible to be within the min and max but still not match one of the case statements?

Reply:
I guess even if there aren't any chips currently supporting bits outside of this range (I'm not sure if there is) the next one released might change this.
The gap in the logic should be filled to be safe.

Reply:
I've reworked this slightly now. I think the logic is a bit clearer now too (with a new comment added)

Reply:
Looks good now!